### PR TITLE
Cleanly terminating Pilot and Ray

### PR DESCRIPTION
**Pull request by drelu**
- Cleanly terminating Ray and Pilot
- Some work on CC mini app

### CODE REVIEW

File: src/mini_apps/quantum_simulation/circuit_cutting/cc_local_cpu.py [hunk 1]
```diff
@@ -12,6 +18,34 @@
...
+            # Check if the process's command line contains the keyword
+            if proc.info["cmdline"] and any(keyword in arg for arg in proc.info["cmdline"]):
+                pid = proc.info["pid"]
+                print(f"Killing process {pid} ({proc.info['name']}) with command: {proc.info['cmdline']}")
+                proc.kill()
+        except (psutil.NoSuchProcess, psutil.AccessDenied):
+            # Process may have terminated or we don't have permissions
+            pass
+
+def stop_ray():
```
Comment (pradeepmantha) [inline src/mini_apps/quantum_simulation/circuit_cutting/cc_local_cpu.py:55]: Pilot-Quantum when starts do ray - stop.. 


https://github.com/radical-cybertools/pilot-quantum/blob/ray-v2/pilot/plugins/ray_v2/cluster.py#L36

Comment (pradeepmantha) [reply src/mini_apps/quantum_simulation/circuit_cutting/cc_local_cpu.py:55]: i am not sure why we need this here again?

File: src/engine/cluster/pilot_quantum_executor.py [hunk 2]
```diff
@@ -25,9 +16,24 @@ def __init__(self, cluster_config=None):
...
+        pilot = pcs.create_pilot(pilot_compute_description=pilot_compute_description)
+        pilot.wait()
+        dask_ray_client = pilot.get_client()
+        return pilot, dask_ray_client
 
     def close(self):
+        if self.type == "dask":
+            self.client.close()
+        elif self.type == "ray":
+            ray.shutdown()
```
Comment (pradeepmantha) [inline src/engine/cluster/pilot_quantum_executor.py:36]: the pilot.cancel which abstracts dask/ray engine should clean it. this seems like leaking pilot-abstraction into mini-apps.

Comment (pradeepmantha) [reply src/engine/cluster/pilot_quantum_executor.py:36]: The pilot-cancel already does this - https://github.com/radical-cybertools/pilot-quantum/blob/ray-v2/pilot/plugins/ray_v2/cluster.py#L136

**Comment by drelu** [comment]
Hi @pradeepmantha, understand that these things should go into Pilot-Quantum. Can you nevertheless merge the branch as it contains substantial further improvements to the circuit cutting mini-app. I suggest to remove the "hacks" after further testing of all mini apps and the paper deadline.

**Pull request merged**